zanata-adapter-xliff/src/main/java/org/zanata/adapter/xliff/XliffWriter.java

```diff
@@ -216,30 +221,12 @@ public static void writeFile(File file, Resource doc, String locale,
             // end xliff tag
             writer.writeEndDocument();
             writer.flush();
+            writer.close();
```

Comment:
You can not put ```writer.close()``` in finally block because FileOutputStream will be closed before it reaches our defined finally (try with resource will close its resources first I guess?). Then it will be an error. Making a Closeable wrapper or subclass will not help because the way the writer is instantiated:
```
try (FileOutputStream fileStream = new FileOutputStream(file)) {
            serializer.setOutputStream(fileStream);
            writer = serializer.getXMLStreamWriter();
```
FileOutputStream is always instantiated first before you can get a writer. But writer must be closed before you close FileOutpuStream. If you must do closing in finally, then we can not use try with resource. The only actual resource is just the FileOutputStream so I think this solution is better than having the old try finally which you will then need to catch exception inside finally.